gcp/src/main/java/org/apache/iceberg/gcp/GCPProperties.java

```diff
@@ -62,6 +62,9 @@ public class GCPProperties implements Serializable {
    */
   public static final int GCS_DELETE_BATCH_SIZE_DEFAULT = 50;
 
+  public static final String GCP_CREDENTIALS_PATH_PROPERTY = "gcp.auth.credentials-path";
```

Comment:
I'm currently not sure whether these properties should be here vs in `GoogleAuthManager`. The properties in this class are typically related to the FileIO implementation of GCS, whereas these new properties aren't